### ISSUE TITLE
Improve aggrMethod

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-sth-comet#432]
Currently:

```
aggrMethod must be one of max, min, sum, sum2, occur
```

It would be great:

1. To include costless additional aggregations. For example, average is sum by occur. It would be very easy for STH to generate avg at rendering time.
2. To be able to include several aggregations at the same time using a comma separeted list, e.g. `aggrMethod=min,max`. The `all` token could be used as alias for "all metrics".

**Comment by MadhuNEC** [comment]
Hi @fgalan,

Could you please confirm if anybody is working on this issue? 

> _2. To be able to include several aggregations at the same time using a comma separated list, e.g. aggrMethod=min,max. The all token could be used as alias for "all metrics"._

As per my understanding, below API should work fine for the above improvement. Please confirm my understanding.
curl -v 'http://localhost:8666/STH/v2/entities/Room1/attrs/temperature?type=Room&aggrMethod=min,max&aggrPeriod=second&dateFrom=2016-01-01T00:00:00.000Z&dateTo=2023-01-01T23:59:59.999Z' -s -S -H 'Content-Type: application/json' -H 'Fiware-Service: qsg' -H 'Fiware-ServicePath: /testsink'

**Comment by MadhuNEC** [comment]
Hi @fgalan ,

Could you please confirm the queries?

**Comment by fgalan** [comment]
As far as I know, nobody is working on this issue so far.

Your query is correct. Note that `?aggrMethod=min,max` is just an example. All combination of existing aggrMethod should be supported, in addition to the keyword `all`, meaning all them.

Thanks!

**Comment by MadhuNEC** [comment]
Hi @fgalan ,

>1. To include costless additional aggregations. For example, average is sum by occur. It would be very easy for STH to generate avg at rendering time.

As per my understanding, `sum` is calculated for **attribute type float** and `occur` is calculated for **attribute type string**. 
So, we need to calculate the avg by different method. Also, it could only be implemented for attribute type float.
Could you please share your opinion?
Could you please share your ideas on more examples of additional aggregations?

**Comment by fgalan** [comment]
> As per my understanding, `sum` is calculated for **attribute type float** and `occur` is calculated for **attribute type string**.
> So, we need to calculate the avg by different method. Also, it could only be implemented for attribute type float.
> Could you please share your opinion?

Yes, maybe `occur` is not the one to use. You have to divide `sum` by the number of summed elements.

>  Could you please share your ideas on more examples of additional aggregations?

Part of the work on this issue is to come up with more examples of additional aggregations :) Do you have any idea regarding this?

**Comment by fgalan** [comment]
PR https://github.com/telefonicaid/fiware-sth-comet/pull/605 implements step 2

> 2. To be able to include several aggregations at the same time using a comma separeted list, e.g. aggrMethod=min,max. The all token could be used as alias for "all metrics".

step 1 is still pending

> 1. To include costless additional aggregations. For example, average is sum by occur. It would be very easy for STH to generate avg at rendering time.
